### ISSUE TITLE
onDelete is not working for belongsToMany

### ISSUE THREAD

**Issue by suchy** [sequelize/sequelize#7149]
Hi, 

I'm trying to deal with removing through table records when deleting the assiociations records.

Category model:
```js
module.exports = function (Sequelize, types) {
  const Category = Sequelize.define('category', {
    name: {...},

    color: {...},

    userId: {...}
  }, {
    classMethods: {
      associate: function(models) {
        Category.belongsToMany(models.Transaction, {
          through: 'categories_transactions', onDelete: 'CASCADE'
        })
      }
    },
  })

  return Category
}
```

Transaction model:
```js
module.exports = function (Sequelize, types) {
  const Transaction = Sequelize.define('transaction', {
    amount: {...},

    description: {... },

    date: {...},

    userId: {...}
  }, {
    classMethods: {
      associate: (models) => {
        Transaction.belongsToMany(models.Category, {
          through: 'categories_transactions', onDelete: 'CASCADE'})
      }
    }
  })

  return Transaction
}
```

Now (using express) I'm removing transaction:
```js
exports.delete = (req, res, next) => {
  Transaction
    .destroy({where: {id: req.params.id, userId: USER_ID}})
    .then((rowsCount) => {
      if (!rowsCount ) throw new NotFoundError()
      res.status(204).send()
    })
    .catch(next)
}
```

Transaction is removed from database, but in `categories_transactions` I can see not removed record. I would expect that records will be removed from  `categories_transactions`.

I'm not sure what I'm doing wrong.

Sequelize version: `3.29.0`
Dialect `MySQL`

Thanks!

**Comment by Aukhan** [comment]
I think you need to pass `cascade: true` in the options object of that destroy call.

**Comment by MatthewAry** [comment]
This issue exists in Sequelize version 4.2 as well. This issue should not be closed by probot until properly addressed.

**Comment by mcranston18** [comment]
Using `4.4.2` and the problem persists. [Here is my test example](https://github.com/mcranston18/sequelize-has-many/blob/master/belongs-to-many.js). 

## Steps to reproduce

1. Create models and `belongsToMany` relationship
```
  const Film = db.define('film', {
    id: {
      type: Sequelize.UUID,
      defaultValue: Sequelize.UUIDV4,
      primaryKey: true
    },
    name: {
      type: Sequelize.STRING
    }
  }, {
    underscored: true
  });

  const Festival = db.define('festival', {
    id: {
      type: Sequelize.UUID,
      defaultValue: Sequelize.UUIDV4,
      primaryKey: true
    },
    name: {
      type: Sequelize.STRING
    }
  }, {
    underscored: true
  });

  const FilmFestival = db.define('film_festivals', {}, {});

  Film.belongsToMany(Festival, {
    through: FilmFestival,
    foreignKey: {
      name: 'film_id',
      allowNull: false
    },
    onDelete: 'CASCADE'
  });

  Festival.belongsToMany(Film, {
    through: FilmFestival,
    foreignKey: {
      name: 'festival_id',
      allowNull: false
    },
    onDelete: 'CASCADE'
  });
```

2. Create and associate model instances
```
models.Film.create({name: 'Aviator'}).then(...);
models.Festival.create({name: 'TIFF'}).then(...);
filmInstance.setFestivals([festivalInstance]).then(...)
```

3. Delete model instance
```
// get film instance of just created film
filmInstance.destroy().then(...);
festivalInstance.destroy().then(...);
```

4. Check DB.  `films` and `festivals` tables will correctly be empty after destroy. but `film_festivals` table will have the one entry from previous association.

**Comment by hehoo** [comment]
Is it fixed?

**Comment by keepcosmos** [comment]
It's not fixed in 5.1.0, please reopen

**Comment by ALiangLiang** [comment]
@hehoo 
https://github.com/sequelize/sequelize/issues/10678#issuecomment-478429303
